python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -358,3 +357,243 @@ def explain_graph(self, graph, feat, tmp=1.0, training=False, **kwargs):
+        r"""Learn and return an edge mask that plays a crucial role to
+        explain the prediction made by the GNN for a graph. Also, return
+        the prediction made with the edges chosen based on the edge mask.
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            A heterogeneous graph.
+        feat : dict[str, Tensor]
+            A dict mapping node types (keys) to feature tensors (values).
+            The input features are of shape :math:`(N_t, D_t)`. :math:`N_t` is the
+            number of nodes for node type :math:`t`, and :math:`D_t` is the feature
+            size for node type :math:`t`
+        tmp : float
+            The temperature parameter fed to the sampling procedure.
+        training : bool
+            Training the explanation network.
+        kwargs : dict
+            Additional arguments passed to the GNN model.
+
+        Returns
+        -------
+        Tensor
+            Classification probabilities given the masked graph. It is a tensor of
+            shape :math:`(B, L)`, where :math:`L` is the different types of label
+            in the dataset, and :math:`B` is the batch size.
+        dict[str, Tensor]
+            A dict mapping edge types (keys) to edge tensors (values) of shape :math:`(E_t)`,
+            where :math:`E_t` is the number of edges in the graph for edge type :math:`t`.
+            A higher weight suggests a larger contribution of the edge.
+
+        Examples
+        --------
+
+        >>> import dgl
+        >>> import torch as th
+        >>> import torch.nn as nn
+        >>> import numpy as np
+
+        >>> # Define the model
+        >>> class Model(nn.Module):
+        ...     def __init__(self, in_feats, hid_feats, out_feats, rel_names):
+        ...         super().__init__()
+        ...         self.conv = dgl.nn.HeteroGraphConv(
+        ...             {rel: dgl.nn.GraphConv(in_feats, hid_feats) for rel in rel_names},
+        ...             aggregate="sum",
+        ...         )
+        ...         self.fc = nn.Linear(hid_feats, out_feats)
+        ...         nn.init.xavier_uniform_(self.fc.weight)
+        ...
+        ...     def forward(self, g, h, embed=False, edge_weight=None):
+        ...         if edge_weight:
+        ...             mod_kwargs = {
+        ...                 etype: {"edge_weight": mask} for etype, mask in edge_weight.items()
+        ...             }
+        ...             h = self.conv(g, h, mod_kwargs=mod_kwargs)
+        ...         else:
+        ...             h = self.conv(g, h)
+        ...
+        ...         if embed:
+        ...             return h
+        ...
+        ...         with g.local_scope():
+        ...             g.ndata["h"] = h
+        ...             hg = 0
+        ...             for ntype in g.ntypes:
+        ...                 hg = hg + dgl.mean_nodes(g, "h", ntype=ntype)
+        ...             return th.sigmoid(self.fc(hg))
```

Comment:
Using `th.sigmoid` here may not be appropriate. In particular, `out_feats` is not necessarily 2.

Reply:
you are right I forgot to fix this, done now.